### PR TITLE
Enable billing for PostgreSQL standbys

### PR DESCRIPTION
**Pull request by byucesoy**
There are few alternative ways for billing PostgreSQL standbys, but none are
perfect.

One option was multiplying the amount with number of standbys, but then, there
is no way to distinguish between 8 vCPU backed database from 4 vCPU backed and
HA enabled database.

Another option was adding a count to the billing records, but that would need
a bigger refactoring that I'm not too keen on right now. Though, I noted this
requirement for billing system v2.

Simple and good enough method is adding new billing rates for standbys. This
also has few extra benefits such as it allows us to charge different rate for
standbys if we want.

### CODE REVIEW

File: config/billing_rates.yml [hunk 1]
```diff
@@ -1,20 +1,26 @@
...
-- { id: e50e6bff-6ee9-493b-bc09-75f122863d08, resource_type: GitHubRunnerMinutes, resource_family: standard-8,      location: global,         unit_price: 0.0032 }
-- { id: 89cda6c2-2650-4ac1-bc28-deeae5e3b756, resource_type: GitHubRunnerMinutes, resource_family: standard-16,     location: global,         unit_price: 0.0064 }
-- { id: ab92083e-2be7-4038-b946-b060af3994fb, resource_type: GitHubRunnerMinutes, resource_family: standard-2-arm,  location: global,         unit_price: 0.0008 }
-- { id: 078d2f47-5827-49fa-9d13-bee6d1d8cadd, resource_type: GitHubRunnerMinutes, resource_family: standard-4-arm,  location: global,         unit_price: 0.0016 }
-- { id: 0fc7f264-87f0-446d-bee7-c0454149b1bc, resource_type: GitHubRunnerMinutes, resource_family: standard-8-arm,  location: global,         unit_price: 0.0032 }
-- { id: 995ab760-5d97-4a7b-981e-c9947233c78e, resource_type: GitHubRunnerMinutes, resource_family: standard-16-arm, location: global,         unit_price: 0.0064 }
+- { id: 139d9a67-8182-8578-a303-235cabd5161c, resource_type: VmCores,                resource_family: standard,        location: hetzner-fsn1,   unit_price: 0.000622768 }
```
Comment (byucesoy) [inline config/billing_rates.yml:1]: Most lines are just whitespace changes as I added spaces for aligning.

Comment (byucesoy) [reply config/billing_rates.yml:1]: This is a better view of the diff; https://www.diffchecker.com/e9TfQDnP/

Comment (enescakir) [reply config/billing_rates.yml:1]: GitHub is also able to hide whitespaces
Screenshot 2024-02-13 at 16 31 06

File: prog/postgres/postgres_resource_nexus.rb [hunk 2]
```diff
@@ -134,21 +134,25 @@ def before_run
...
+
+      resource_type = (index == 0) ? "PostgresStorage" : "PostgresStandbyStorage"
+      BillingRecord.create_with_id(
+        project_id: postgres_resource.project_id,
+        resource_id: postgres_resource.id,
+        resource_name: postgres_resource.name,
+        billing_rate_id: BillingRate.from_resource_properties(resource_type, "standard", postgres_resource.location)["id"],
+        amount: postgres_resource.target_storage_size_gib
+      )
+    end
```
Comment (furkansahin) [inline prog/postgres/postgres_resource_nexus.rb:155]:
```suggestion
    (postgres_resource.required_standby_count + 1).times do |index|
       ["Cores", "Storage"].each do |type|
         resource_type = (index == 0) ? "Postgres#{type}" : "PostgresStandby#{type}"
         BillingRecord.create_with_id(
           project_id: postgres_resource.project_id,
           resource_id: postgres_resource.id,
           resource_name: postgres_resource.name,
           billing_rate_id: BillingRate.from_resource_properties(resource_type, "standard", postgres_resource.location)["id"],
           amount: representative_server.vm.cores
         )
       end
     end
```

File: config/billing_rates.yml (hunk 1 above)
Comment (byucesoy) [reply config/billing_rates.yml:1]: I was looking for this option thinking it is such a core feature, it must be somewhere, but couldn't find it :smile:

File: prog/postgres/postgres_resource_nexus.rb (hunk 2 above)
Comment (byucesoy) [reply prog/postgres/postgres_resource_nexus.rb:155]: I actually prefer the current version as it is more readable. I love shaving off some lines, but looping through array of array and generating resource_type using string concatenation seemed bit too much to me.

Comment (fdr) [reply prog/postgres/postgres_resource_nexus.rb:155]: There's a lot of common subexpression in your version though, is there a version without sub-string sp;licing that doesn't repeat, like, 

```
        project_id: postgres_resource.project_id,
        resource_id: postgres_resource.id,
        resource_name: postgres_resource.name,
```

among other content?

Reading the code to confirm that these are actually held in common is a bit of an eye-maze.

Comment (byucesoy) [reply prog/postgres/postgres_resource_nexus.rb]: I still prefer the old one, but I wrote a version that does not duplicate BillingRecord.create_with_id calls. How about this?

Comment (furkansahin) [reply prog/postgres/postgres_resource_nexus.rb]: I am OKish for the new version. To be honest, I personally prefer having the array written out as I have provided in the suggestion, but this is a personal preference. I leave the decision to you and @fdr

**Review by furkansahin — APPROVED** [review]

**Pull request merged**